public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.tsx

```diff
@@ -1,68 +1,54 @@
-import React from 'react';
-import { AppEvents } from '@grafana/data';
+import React, { useState } from 'react';
+import { AppEvents, PluginType } from '@grafana/data';
 import { Button, HorizontalGroup, useStyles2 } from '@grafana/ui';
 import appEvents from 'app/core/app_events';
-import { api } from '../../api';
-import { ActionTypes, CatalogPlugin, PluginStatus } from '../../types';
+
+import { CatalogPlugin, PluginStatus } from '../../types';
 import { getStyles } from './index';
+import { useInstallStatus, useUninstallStatus, useInstall, useUninstall } from '../../state/hooks';
 
 type InstallControlsButtonProps = {
-  isInProgress: boolean;
-  hasInstalledPanel: boolean;
-  dispatch: React.Dispatch<any>;
   plugin: CatalogPlugin;
   pluginStatus: PluginStatus;
 };
 
-export function InstallControlsButton({
-  isInProgress,
-  dispatch,
-  plugin,
-  pluginStatus,
-  hasInstalledPanel,
-}: InstallControlsButtonProps) {
-  const uninstallBtnText = isInProgress ? 'Uninstalling' : 'Uninstall';
-  const updateBtnText = isInProgress ? 'Updating' : 'Update';
-  const installBtnText = isInProgress ? 'Installing' : 'Install';
+export function InstallControlsButton({ plugin, pluginStatus }: InstallControlsButtonProps) {
+  const { isInstalling, error: errorInstalling } = useInstallStatus();
+  const { isUninstalling, error: errorUninstalling } = useUninstallStatus();
+  const install = useInstall();
+  const uninstall = useUninstall();
+  const [hasInstalledPanel, setHasInstalledPanel] = useState(false);
   const styles = useStyles2(getStyles);
+  const uninstallBtnText = isUninstalling ? 'Uninstalling' : 'Uninstall';
 
   const onInstall = async () => {
```

Comment:
Nit: could be a good idea to wrap this in a `useAyncFn` to prevent the risk of calling `setHasInstalledPanel` after the component has unmounted.

Reply:
The same thing goes for the other async functions in this component.

Reply:
Yea, I think that's a fair point, will change 👍